src/array.rs

```diff
@@ -72,6 +72,19 @@ impl Array {
     state.array.swap(a, b);
   }
 
+//  /// Reverses the Array.
+//  pub fn reverse(&self) {
+//    let mut state = self.0.get();
+//    state.array.reverse();
+//  }
```

Comment:
Is this code needed for the PR?

Reply:
Not needed, but relevant. It will be needed when implementing different reset mechanics between each algorithm in all algorithm mode.